### PR TITLE
Gyro example bot

### PR DESCRIPTION
**Pull request by Twinters007**

### CODE REVIEW

File: examples/examples/Gyro/robot.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'''
+Created on Jan 10, 2016
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Please remove the top comment.

File: examples/examples/Gyro/robot.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    
+        #gyro calibration constant, may need to be adjusted; 
+        #gyro value of 360 is set to correspond to one full revolution
+        self.voltsPerDegreePerSecond = .0128
+              
+        self.myRobot = wpilib.RobotDrive(wpilib.CANTalon(self.leftMotorChannel), wpilib.CANTalon(self.leftRearMotorChannel), wpilib.CANTalon(self.rightMotorChannel), wpilib.CANTalon(self.rightRearMotorChannel))
+        self.gyro = wpilib.Gyro(self.gyroChannel)
+        self.joystick = wpilib.Joystick(self.joystickChannel)
+    
+        #Runs during autonomous
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: This should be a docstring.

File: examples/examples/Gyro/robot.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        
+        self.joystickChannel = 0; #usb number in DriverStation
+    
+        #hannels for motors
+        self.leftMotorChannel = 1
+        self.rightMotorChannel = 0
+        self.leftRearMotorChannel = 3
+        self.rightRearMotorChannel = 2
+    
+        self.angleSetpoint = 0.0;
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Semicolons

File: examples/examples/Gyro/robot.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        self.myRobot = wpilib.RobotDrive(wpilib.CANTalon(self.leftMotorChannel), wpilib.CANTalon(self.leftRearMotorChannel), wpilib.CANTalon(self.rightMotorChannel), wpilib.CANTalon(self.rightRearMotorChannel))
+        self.gyro = wpilib.Gyro(self.gyroChannel)
+        self.joystick = wpilib.Joystick(self.joystickChannel)
+    
+        #Runs during autonomous
+    def autonomous(self):
+        pass
+
+    '''
+     Sets the gyro sensitivity and drives the robot when the joystick is pushed. The
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Docstring goes below `def operatorControl`

File: examples/examples/Gyro/robot.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        while self.isOperatorControl() and self.isEnabled():
+            turningValue =  (self.angleSetpoint - self.gyro.getAngle())*self.pGain
+            if self.joystick.getY() <= 0:
+                #forwards
+                self.myRobot.drive(self.joystick.getY(), turningValue)
+            elif self.joystick.getY() >0:
+                #backwards
+                self.myRobot.drive(self.joystick.getY(), -turningValue)
+            wpilib.Timer.delay(0.005)
+    #Runs during test mode
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Please add spacing between functions.

File: examples/examples/Gyro/robot.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    def test(self):
+        pass
+if __name__ == "__main__":
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Ditto.

File: examples/examples/Gyro/robot.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+     '''
+        
+    def operatorControl(self):
+        self.gyro.setSensitivity(self.voltsPerDegreePerSecond) #calibrates gyro values to equal degrees
+        while self.isOperatorControl() and self.isEnabled():
+            turningValue =  (self.angleSetpoint - self.gyro.getAngle())*self.pGain
+            if self.joystick.getY() <= 0:
+                #forwards
+                self.myRobot.drive(self.joystick.getY(), turningValue)
+            elif self.joystick.getY() >0:
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Should be a space between `>` and `0`

File: examples/examples/Gyro/robot.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    
+        self.angleSetpoint = 0.0;
+        self.pGain = 1; #propotional turning constant
+    
+        #gyro calibration constant, may need to be adjusted; 
+        #gyro value of 360 is set to correspond to one full revolution
+        self.voltsPerDegreePerSecond = .0128
+              
+        self.myRobot = wpilib.RobotDrive(wpilib.CANTalon(self.leftMotorChannel), wpilib.CANTalon(self.leftRearMotorChannel), wpilib.CANTalon(self.rightMotorChannel), wpilib.CANTalon(self.rightRearMotorChannel))
+        self.gyro = wpilib.Gyro(self.gyroChannel)
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: There is no Gyro object in 2016, it's called AnalogGyro. Did you run this with the 2016 version of robotpy?

**Comment by virtuald** [comment]
Please add it to the list of tests at https://github.com/robotpy/robotpy-wpilib/blob/master/wpilib/tests/run_tests.sh#L29

File: examples/examples/Gyro/robot.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        self.rightRearMotorChannel = 2
+    
+        self.angleSetpoint = 0.0;
+        self.pGain = 1; #propotional turning constant
+    
+        #gyro calibration constant, may need to be adjusted; 
+        #gyro value of 360 is set to correspond to one full revolution
+        self.voltsPerDegreePerSecond = .0128
+              
+        self.myRobot = wpilib.RobotDrive(wpilib.CANTalon(self.leftMotorChannel), wpilib.CANTalon(self.leftRearMotorChannel), wpilib.CANTalon(self.rightMotorChannel), wpilib.CANTalon(self.rightRearMotorChannel))
```
Comment (virtuald) [inline examples/examples/Gyro/robot.py]: Also, can you split this to multiple lines. It's pretty ugly as a single line.

**Comment by virtuald** [comment]
The only reason I'm being so picky is that if this is supposed to be an example for newbies to follow, it should be high quality. :)

**Comment by Twinters007** [comment]
For some reason the run_tests.sh file cuts off after the pythonpath. Any reason why this would happen?

**Comment by virtuald** [comment]
Cuts off? What are you using to edit it?

**Comment by Twinters007** [comment]
Atom

**Comment by Twinters007** [comment]
Same when I use eclipse' text editor

**Comment by virtuald** [comment]
Maybe you accidentally clipped it. You can use `git status` to see if there are changes to the file, and `git checkout filename` to restore it to its original contents.

**Comment by Twinters007** [comment]
it's in wpilib/tests right?

**Comment by Twinters007** [comment]
This is the output of `cat` after I checkout

```#!/bin/bash

set -e
cd `dirname $0`

PYTHONPATH=..:../../hal-sim:../../hal-base python3 -m coverage run --source wpilib -m pytest $@
python -m coverage report -m
```

**Comment by Twinters007** [comment]
Ok...I know what happened. I forked this repo last year. So I'm 168 commits behind. Let me fix this

**Pull request closed without merging**